src/features/loading.ts

```diff
@@ -382,30 +407,39 @@ export const LoadingMixin = (ModelViewerElement:
           const defaultPosterElement = this[$defaultPosterElement];
 
           if (posterContainerElement.classList.contains('show')) {
+            const finishHidingPoster = () => {
+              // Ensure that the poster is no longer focusable or visible to
+              // screen readers
+              defaultPosterElement.setAttribute('aria-hidden', 'true');
+              defaultPosterElement.removeAttribute('tabindex');
+            };
             posterContainerElement.classList.remove('show');
 
-            // We might need to forward focus to our internal canvas, but that
-            // cannot happen until the poster has completely transitioned away
-            posterContainerElement.addEventListener('transitionend', () => {
-              this[$announceModelVisibility](true);
-              requestAnimationFrame(() => {
-                this[$modelIsVisible] = true;
-
-                const root = this.getRootNode();
-
-                // If the <model-viewer> is still focused, forward the focus to
-                // the canvas that has just been revealed
-                if (root &&
-                    (root as Document | ShadowRoot).activeElement === this) {
-                  this[$canvas].focus();
-                }
-
-                // Ensure that the poster is no longer focusable or visible to
-                // screen readers
-                defaultPosterElement.setAttribute('aria-hidden', 'true');
-                defaultPosterElement.removeAttribute('tabindex');
-              });
-            }, {once: true});
+            if (this[$elementIsRenderable]) {
+              // We might need to forward focus to our internal canvas, but that
+              // cannot happen until the poster has completely transitioned away
+              posterContainerElement.addEventListener('transitionend', () => {
+                requestAnimationFrame(() => {
+                  this[$changeModelVisibilityState](true);
+
+                  const root = this.getRootNode();
+
+                  // If the <model-viewer> is still focused, forward the focus
+                  // to the canvas that has just been revealed
+                  if (root &&
+                      (root as Document | ShadowRoot).activeElement === this) {
+                    this[$canvas].focus();
+                  }
+
+                  finishHidingPoster();
+                });
+              }, {once: true});
+            } else {
+              // For non-renderable cases, we don't need to bother focusing the
+              // canvas because it cannot be focused anyway
+              this[$changeModelVisibilityState](true);
```

Comment:
What does it mean for an element to be not renderable, but for its model to be visible?

Reply:
That's a good question. I don't think we have been very crisp on this. Perhaps the element being in a non-renderable state should always imply that the model is not visible.

OTOH it seems like a valid case for someone to want to wait until the model would actually be rendered to a canvas before changing `display: none => block`.

Reply:
Personally, as I alluded to elsewhere, I would not expect the model to be visible unless it's being displayed to the user. We don't check that at all here.

Reply:
Yeah, for sure. I think the intention and the name don't really line up well here. I think the intention is something along the lines of, "if the element was rendered, the model would be displayed." The boolean is intended to capture the difference between the canvas been obscured by the poster vs. the poster being hidden.